src/Widgets/CameraView.vala

```diff
@@ -127,6 +127,10 @@ public class Camera.Widgets.CameraView : Gtk.Box {
             main_widget.visible_child = no_device_view;
         } else {
             change_camera (monitor.get_devices ().nth_data (0));
+            ((Camera.MainWindow) this.get_toplevel ()).change_action_state (
+                Camera.MainWindow.ACTION_CHANGE_CAMERA,
+                new Variant.string (monitor.get_devices ().nth_data (0).name)
+            );
```

Comment:
We can call `change_action_state()` as the last thing in `change_camera()` and handle all state changes there. That makes sure we are always setting the right camera name in the state.